tools/swift-format/Sources/Rules/MaximumBlankLines.swift

```diff
@@ -16,5 +16,69 @@ import SwiftSyntax
 ///
 /// - SeeAlso: https://google.github.io/swift#vertical-whitespace
 public final class MaximumBlankLines: SyntaxFormatRule {
+  public override func visit(_ token: TokenSyntax) -> Syntax {
+    guard let parentToken = token.parent else {
+      return token.withLeadingTrivia(removeExtraBlankLines(token).0)
+    }
 
+    guard let grandparentTok = parentToken.parent else {
+      return token.withLeadingTrivia(removeExtraBlankLines(token).0)
+    }
+
+    // Tokens who appeared in a member type are handle by
+    // BlankLineBetweenMembers rule.
+    if grandparentTok is MemberDeclListItemSyntax {
+      return token
+    }
+    else {
+      let correctTrivia = removeExtraBlankLines(token)
+      return correctTrivia.1 ? token : token.withLeadingTrivia(removeExtraBlankLines(token).0)
+    }
+  }
+
+  /// Indicates if the given trivia contains an invalid amount of
+  /// consecutively blank lines. If it does it returns a clean trivia
+  /// with the correct amount of blank lines.
+  func removeExtraBlankLines(_ token: TokenSyntax) -> (Trivia, Bool) {
```

Comment:
Use a labeled tuple instead of an unlabeled one, so that the usage is cleaner at the call site. For example, define this to return something like `... -> (newTrivia: Trivia, hasValidAmountOfBlankLines: Bool)` and then you can replace the `.0` and `.1` above with those labels.